### ISSUE TITLE
master: can't log in on setup.pl anymore without a company name

### ISSUE THREAD

**Issue by ehuelsmann** [ledgersmb/LedgerSMB#2281]
When hitting 'login' on setup.pl with a valid user and password, a list of allowable company names should appear. @sbts  says that's no longer the case. Registering this bug for him to look at tomorrow.

**Comment by JigmeDatse** [comment]
I am just looking at it right now.  The error is very uninformative:

```
Use of uninitialized value in concatenation (.) or string at /home/jrasku/src/ledgersmb/LedgerSMB/tools/../lib/LedgerSMB/PSGI.pm line 171.
```

**Comment by JigmeDatse** [comment]
`use Carp::Always` helps somewhat:

```
Use of uninitialized value in concatenation (.) or string at /home/jrasku/src/ledgersmb/LedgerSMB/tools/../lib/LedgerSMB/PSGI.pm line 170.
	LedgerSMB::PSGI::catch {...} ("Can't use an undefined value as an ARRAY reference at /home/j"...) called at /usr/local/lib64/perl5/5.22.2/Try/Tiny.pm line 124
	Try::Tiny::try(CODE(0xb087f68), Try::Tiny::Catch=REF(0xb0e4678)) called at /home/jrasku/src/ledgersmb/LedgerSMB/tools/../lib/LedgerSMB/PSGI.pm line 182
	LedgerSMB::PSGI::psgi_app(HASH(0xa77cfd8)) called at /usr/local/lib64/perl5/5.22.2/Plack/App/URLMap.pm line 71
	Plack::App::URLMap::call(Plack::App::URLMap=HASH(0xa8d94c8), HASH(0xa77cfd8)) called at /usr/local/lib64/perl5/5.22.2/Plack/Component.pm line 50
	Plack::Component::__ANON__(HASH(0xa77cfd8)) called at /usr/local/lib64/perl5/5.22.2/Plack/Middleware/Conditional.pm line 16
	Plack::Middleware::Conditional::call(Plack::Middleware::Conditional=HASH(0xa8e8be8), HASH(0xa77cfd8)) called at /usr/local/lib64/perl5/5.22.2/Plack/Component.pm line 50
	Plack::Component::__ANON__(HASH(0xa77cfd8)) called at /usr/local/lib64/perl5/5.22.2/Plack/Middleware/Redirect.pm line 73
	Plack::Middleware::Redirect::call(Plack::Middleware::Redirect=HASH(0xa8e8f60), HASH(0xa77cfd8)) called at /usr/local/lib64/perl5/5.22.2/Plack/Component.pm line 50
	Plack::Component::__ANON__(HASH(0xa77cfd8)) called at /usr/local/lib64/perl5/5.22.2/Plack/Util.pm line 145
	eval {...} called at /usr/local/lib64/perl5/5.22.2/Plack/Util.pm line 145
	Plack::Util::run_app(CODE(0xa8e8ae0), HASH(0xa77cfd8)) called at /usr/local/lib64/perl5/5.22.2/Starman/Server.pm line 195
	Starman::Server::dispatch_request(Starman::Server=HASH(0x26c2c10), HASH(0xa77cfd8)) called at /usr/local/lib64/perl5/5.22.2/Starman/Server.pm line 298
	Starman::Server::process_request(Starman::Server=HASH(0x26c2c10), Net::Server::Proto::TCP=GLOB(0xa7e1ef0)) called at /usr/local/lib64/perl5/5.22.2/Net/Server.pm line 74
	Net::Server::run_client_connection(Starman::Server=HASH(0x26c2c10)) called at /usr/local/lib64/perl5/5.22.2/Net/Server/PreFork.pm line 229
	eval {...} called at /usr/local/lib64/perl5/5.22.2/Net/Server/PreFork.pm line 229
	Net::Server::PreFork::run_child(Starman::Server=HASH(0x26c2c10)) called at /usr/local/lib64/perl5/5.22.2/Net/Server/PreFork.pm line 187
	Net::Server::PreFork::run_n_children(Starman::Server=HASH(0x26c2c10), 5) called at /usr/local/lib64/perl5/5.22.2/Net/Server/PreFork.pm line 111
	Net::Server::PreFork::loop(Starman::Server=HASH(0x26c2c10)) called at /usr/local/lib64/perl5/5.22.2/Net/Server.pm line 60
	Net::Server::run(Starman::Server=HASH(0x26c2c10), "port", ARRAY(0x2c30060), "host", "*", "proto", "tcp", "serialize", "none", ...) called at /usr/local/lib64/perl5/5.22.2/Starman/Server.pm line 106
	Starman::Server::run(Starman::Server=HASH(0x26c2c10), CODE(0x26c2b68), HASH(0x26c2dc0)) called at /usr/local/lib64/perl5/5.22.2/Plack/Handler/Starman.pm line 25
	Plack::Handler::Starman::run(Plack::Handler::Starman=HASH(0x26c2b50), CODE(0x26c2b68)) called at /usr/local/lib64/perl5/5.22.2/Plack/Loader/Delayed.pm line 20
	Plack::Loader::Delayed::run(Plack::Loader::Delayed=HASH(0x25f3088), Plack::Handler::Starman=HASH(0x26c2b50)) called at /usr/local/lib64/perl5/5.22.2/Plack/Runner.pm line 277
	Plack::Runner::run(Plack::Runner=HASH(0x25b80c8)) called at /usr/local/bin/starman line 38
```

**Comment by ylavoie** [comment]
Fixed by PR#2286. We had to ensure the list of databases was returned upward.
